### DISCUSSION TITLE
mitmproxy encodes get-request parameter that's an url

### DISCUSSION THREAD

**Discussion by tobiasmueller** [mitmproxy/mitmproxy#5812] [Q&A] [answered]
mitmproxy encodes get-request parameter that's an url

STEPS TO REPRODUCE
	1. Install official mitmproxy on Windows 11
	2. Modify a get request with an url as query parameter with an addon

Original get request:
https://oauth2.example.net/authorize?client_id=example_extension_ff&redirect_uri=https://698bf.extensions.allizom.org/

Addon code:
(Goal: bring everything to a QA stage)

```
def request(flow):
    if flow.request.host == "oauth2.example.net":
        flow.request.host = "oauth2-qa.example.net"
        flow.request.query["client_id"] = "example_extension_ff_qa"
```

Current result:
https://oauth2-qa.example.net/authorize?client_id=example_extension_ff_qa&redirect_uri=https%3A%2F%2F698bf.extensions.allizom.org

Expected result:
https://oauth2-qa.example.net/authorize?client_id=example_extension_ff_qa&redirect_uri=https://698bf.extensions.allizom.org/

How can I prevent the encoding of the redirect_uri parameter?

**Comment by Prinzhorn** [comment, marked as answer]
If you want different behavior you need to manually edit `request.path` without going through `request.query` (which is basically a convenient proxy to manipulate parts of `path`).

The default implementation correctly encodes [reserved characters](https://datatracker.ietf.org/doc/html/rfc3986#section-2.2) via `url.encode`. I wouldn't "pick and choose" which characters you personally prefer to be encoded as this can have unwanted and potentially disastrous side-effects.

**Reply by tobiasmueller** [reply]
Thank you for answering
